core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -224,6 +226,7 @@ public void initialize(String name, Map<String, String> unresolved) {
               client, tokenRefreshExecutor(), token, expiresAtMillis(mergedProps), catalogAuth);
     }
 
+    this.restPageSize = mergedProps.get(REST_PAGE_SIZE);
```

Comment:
We should probably add a validation that this is a positive integer if set.

Reply:
will fix this

Reply:
@rahil-c please also add a test that verifies this behavior

Reply:
Will do so